src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Extensions/TimeSpanExtensions.cs

```diff
@@ -32,5 +32,27 @@ public static TimeSpan CalculateBackoffTime(this TimeSpan interval, int attempts
 
             return TimeSpan.FromTicks(Math.Min(maxBackoff.Ticks, calculatedBackoff.Ticks));
         }
+
+        /// <summary>
+        /// This method calculates a random exponential time to retry the primary store after a failure
+        /// which lies between <paramref name="interval"/> and <see cref="RefreshConstants.DefaultMaxRetryAfter"/>.
+        /// </summary>
+        /// <param name="interval">The minimum interval to retry after.</param>
+        /// <param name="attempts">The number of attempts made to the primary config store.</param>
+        /// <returns>The calculated exponential time to retry the primary config store after a failure.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// An exception is thrown when <paramref name="attempts"/> is less than 1.
+        /// </exception>
+        public static TimeSpan CalculateRetryAfterTime(this TimeSpan interval, int attempts)
```

Comment:
>CalculateRetryAfterTime [](http://example.com/codeflow?start=31&length=23)

Can we reuse the existing `CalculateBackoffTime` logic for replica failover too?

Reply:
These two methods are quite different. `CalculateBackoffTime` would wait minimum time before retrying again, and this method will wait for maximum of 10 mins before retrying again. We could potentially make them generic if there are more cases where this type of logic is required, but at this point, I see no need to generalize it.

Reply:
I think we confused ourselves. The fail-over doesn't initiate a request. Fail-over only determines which replica to use for the next request when it comes. The requests are initiated by startup and by refresh. They already have intervals and backoff logic. So there is nothing for the fail-over to back off for. We just need to decide how soon we want to start over and try all the endpoints starting with the first one. For this, I think 10 minutes is a reasonable choice.

Reply:
I think this relates to the 'last known good replica' design and it is not relevant anymore. Correct me if I am wrong.